test/distributed/_tensor/test_random_ops.py

```diff
@@ -263,6 +245,60 @@ def test_deterministic_uniform_2d(self):
                 else:
                     self.assertNotEqual(local_tensor_gathered[slice_idx], local_tensor)
 
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    def test_meta_tensor_init(self):
+        # test suite sets each rank's seed to the same value but in actual
+        # execution the default random seed will be different (a random value).
+        torch.cuda.manual_seed(self.rank)
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        size = [1024, 2048]
+        meta_dtensor = distribute_tensor(
+            torch.empty(*size, device="meta"), device_mesh, [Replicate()]
+        )
+        self.assertTrue(meta_dtensor.is_meta)
+        dtensor = torch.empty_like(meta_dtensor, device=self.device_type)
+
+        # disable the distribute region for RNG
+        random._rng_tracker.distribute_region_enabled = False
+        dtensor.uniform_()
+
+        # allgather the local tensors
+        dtensor = DTensor.from_local(dtensor.to_local(), device_mesh, [Shard(0)])
+        dtensor = dtensor.redistribute(device_mesh, [Replicate()])
+        local_tensor = dtensor.to_local()
```

Comment:
hmm maybe we should call `device_mesh.allgather(dtensor.to_local())` directly instead of creating a sharded then converting to replicated? It's kind of weird if the only thing we want to do here is to allgather